exchanges/request/request.go

```diff
@@ -54,6 +55,10 @@ func (r *Requester) SendPayload(ctx context.Context, ep EndpointLimit, newReques
 		return errRequestSystemIsNil
 	}
 
+	if ctx == nil {
+		return errContextRequired
+	}
+
 	defer r.timedLock.UnlockIfLocked()
 
 	if newRequest == nil {
```

Comment:
Not a change request here, but I think if we were going to check for context expiry GCT side, it should be in `request.go` and it will cover the majority of cases. While the database could also use context pass arounds, I don't think its a high priority right now.

Reply:
Sounds good